armi/settings/caseSettings.py

```diff
@@ -368,13 +368,62 @@ def writeToYamlFile(self, fName, style="short"):
             the current state of settings
         """
         self.path = pathTools.armiAbsPath(fName)
+        if style == "medium":
+            originalSettings = self.getOriginalCaseSettings(self.path)
+        else:
+            originalSettings = None
         with open(self.path, "w") as stream:
-            writer = self.writeToYamlStream(stream, style)
+            writer = self.writeToYamlStream(stream, style, originalSettings)
         return writer
 
-    def writeToYamlStream(self, stream, style="short"):
-        """Write settings in yaml format to an arbitrary stream."""
-        writer = settingsIO.SettingsWriter(self, style=style)
+    def getOriginalCaseSettings(self, fPath):
```

Comment:
I decided to have this live here, and pass the resulting list of "old" settings names into the instantiation of the `SettingsWriter` object (this happens later on in `writeToYamlStream`). This is because the file path names are here and my choice was either to pass in this list or pass the file name and have this method there. The downside to having it here is there must be logic about the style choice in this class too, which might not be desirable. See line 371 above:

https://github.com/terrapower/armi/pull/904/files#diff-0dc0a268887c52884d0cc28261cb1c576277485a38433172ef5bfc4914549780R371-R374

Happy to change the name and name of the original settings names variable if someone has a better suggestion.

Reply:
I might change the name of `originalSettings` and `getOriginalCaseSettings()` to something like `settingsSetByUser` and `getSettingsSetByUser()`.

I'm not super wild about that name either, so do whatever you think is best here. I just think `original` could be interpreted in a number of ways.